contents/docs/contribute/developing-locally.mdx

```diff
@@ -9,231 +9,268 @@ showTitle: true
+python3 -m venv env
 ```
 
-7. Run migrations:
+4. Activate the virtual environment:
 
 ```bash
-DEBUG=1 python3 manage.py migrate
+# for bash/zsh/etc
+source env/bin/activate
+
+# if you're using the fish shell
+source env/bin/activate.fish
 ```
 
-> **Friendly tip:** The error `fe_sendauth: no password supplied` connecting to Postgres happens when the database is set up with a password and the user:pass isn't specified in `DATABASE_URL`. Try `export DATABASE_URL=postgres://posthog:posthog@localhost:5432/posthog`.
+5. Install requirements with pip
 
-> **Another friendly tip:** You may run into `psycopg2` errors while migrating on a Apple Silicon machine. Try out the steps in this [comment](https://github.com/psycopg/psycopg2/issues/1216#issuecomment-820556849) to resolve this. 
-    
-8. Make sure you have [Yarn installed](https://classic.yarnpkg.com/en/docs/install/):
+If you're on Apple Silicon, the first time your run `pip install` you must pass it custom openssl headers:
 
 ```bash
-# macOS (Homebrew)
-brew install yarn
+brew install openssl
+CFLAGS="-I /opt/homebrew/opt/openssl/include" LDFLAGS="-L /opt/homebrew/opt/openssl/lib" GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1 GRPC_PYTHON_BUILD_SYSTEM_ZLIB=1 pip install -r requirements.txt
 ```
 
-9. Make sure you have [NodeJS installed](https://nodejs.org/en/download/):
+These will be used when installing `grpcio` and `psycopg2`. After doing this once, and assuming nothing changed with these two packages, next time simply run:
 
 ```bash
-# macOS (Homebrew)
-brew install node
+pip install -r requirements.txt
 ```
 
-> **Note:** For Apple Silicon support, please use NodeJS >= v15.0
+6. SAML support.
 
-10. Start the backend, worker, and frontend simultaneously with:
-    ```bash
-    ./bin/start
-    ```
+If you want to fully test all our features, you'll need to install a few dependencies for SAML to run properly. If you're on macOS, run the command below, otherwise check out the official [xmlsec repo](https://github.com/mehcode/python-xmlsec) for more details.
 
-> **Note:** The first time you run this command you might get an error that says "layout.html is not defined". Make sure you wait until the frontend is finished compiling and try again.\*
-
-> **Friendly tip:** If you run into frontend webpack module errors running `rm -rf node_modules/` might help.
-
-Now open [http://localhost:8000](http://localhost:8000) to see the app.
+```
+brew install libxml2 libxmlsec1 pkg-config
+pip install python3-saml==1.12.0
+```
 
-To see some data on the frontend, you should go to the `http://localhost:8000/demo` and play around with it, so you can see some data on dashboard.
+7. Install dev requirements:
 
-> **Friendly tip:** Homebrew services can be stopped with `brew services stop <service_name>`
+```bash
+pip install -r requirements-dev.txt
+```
 
-</HiddenSection>
+### Step 5. Run database migrations
 
-<HiddenSection headingType='h2' title='Using Docker'>
+```bash
+export DEBUG=1
```

Comment:
If you use https://github.com/eknkc/fish-autoenv like me, you can add these to your .env.fish not to have to type these out every time :)